### DISCUSSION TITLE
Pressing `Shift` key on a hardware keyboard (official Samsung keyboard cover) is deleting all characters on the left

### DISCUSSION THREAD

**Discussion by alexanderadam** [AppGround-io/bluetooth-keyboard-and-mouse-support#217] [Bugs & Problems]
I bought this app many years ago and this bug has always bothered me.
This happened with various Keyboard covers.

If I try to type a capital letter by pressing shift and the letter on the hardware keyboard, Serverless Bluetooth Keyboard & Mouse Premium will delete everything on the left side of the cursor.
It's basically like pressing backspace all the time.
It works perfectly fine if I'm using the software keyboard but what's the point of having to detach the hardware keyboard every time I'm starting the app?

Currently I'm using Serverless Bluetooth Keyboard & Mouse Premium 4.7.1 but I'm pretty sure there wasn't a version without this bug yet.

**Comment by AppGround-io** [comment]
I tested it with a Samsung Galaxy Tap S7+ with Android 12 and a Samsung Book Cover Keyboard and the problem does not happen on my device.
To reproduce this problem, it would be good to know the exact hardware you are using and the Android version on your device and the keyboard language selected in the app.

**Reply by alexanderadam** [reply]
I'm having the exact tablet with the same cover.
I'm always using US layout though.
